### PR TITLE
Entrypoints

### PR DESCRIPTION
**Pull request by Roynecro97**
* Add `micromouse.robot` entry point for registering additional robots.
* Add `register_robot()` decorator for registering additional robots.
* Add `micromouse.gui` entry point for registering alternative GUI engines (maybe even TUI engines).
* Add `micromouse.tool` entry point for registering additional tools.
* Add `better_random`, a random robot that goes back only if necessary and doesn't drive in reverse. (Example for `micromouse.robot` entry point.)
* Modify `GUIRenderer` to use the `micromouse.gui` entry point.
* Add functions that rotate/transpose/flip `Walls` (and tests).
* Add `MazeEditor` example tool that can render/rotate/transpose/flip/generate mazes.

### CODE REVIEW

File: sim/tools.py [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+    ascii_charset,
+    Charset,
+    Maze,
+    utf8_charset,
+    Walls,
+)
+
+
+MazeFormat = Literal['maze', 'maz', 'num', 'csv']
+MAZE_FORMATS = frozenset({'maze', 'maz', 'num', 'csv'})
```
Comment (OmerHuly) [inline sim/tools.py]: Not sure where those two are used...

File: sim/tools.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+
+class MazeEditor(Tool):
+    """Utilities for viewing an manipulating a maze."""
```
Comment (OmerHuly) [inline sim/tools.py]:
```suggestion
    """Utilities for viewing and manipulating a maze."""
```

File: sim/front.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
+"""Utilities for using frontends this package.
```
Comment (OmerHuly) [inline sim/front.py]: I didn't understand the English in this sentence... 😅

File: sim/tools.py (hunk 1 above)
Comment (Roynecro97) [reply sim/tools.py]: In deleted code obviously... 🙄

`MazeFormat` is still referenced in some documentation that also survived deletion.

File: sim/front.py (hunk 3 above)
Comment (Roynecro97) [reply sim/front.py]: I don't know, functions for frontends... resolve this conversation if I make sense in the documentation now (I'm open to suggestions if not).

**Review by OmerHuly — APPROVED** [review]

**Pull request merged**